CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+cmake_minimum_required(VERSION 3.15...3.27)
+project(${SKBUILD_PROJECT_NAME} LANGUAGES CXX)
+
+message(STATUS ${SKBUILD_PROJECT_NAME})
+
+set(PYBIND11_NEWPYTHON ON)
+find_package(pybind11 CONFIG REQUIRED)
+
+pybind11_add_module(goals_proj src/goals_proj/goals_proj.cpp)
+target_compile_features(goals_proj PRIVATE cxx_std_20)
+
+include(FetchContent)
+message(CHECK_START "Locating GoalsARM C++")
+list(APPEND CMAKE_MESSAGE_INDENT "  ")
+
+set(GOALS_ARM_PATH "Path to local GoalsARM C++ library" CACHE STRING "")
+
+## Stop GoalsARM building tests
+option(GOALS_BUILD_TESTING "Build GoalsARM tests" OFF)
+message(STATUS "Building GoalsARM tests: ${GOALS_BUILD_TESTING}")
+include(CTest)
+
+if(DEFINED ENV{LOCAL_BUILD} AND "$ENV{LOCAL_BUILD}" STREQUAL "true")
```

Comment:
Important bit in this CMake file is that we can switch between a local GoalsARM (for if we want to debug in development work) and GoalsARM from GitHub.